### PR TITLE
Qt: Fix overriding buttons setting when option is null.

### LINKED ISSUE #1965: Game name and framerate aren't shown when the display driver is set to software QT
Capture

Capture2

### PR DESCRIPTION
**Pull request by Testsr**
fixes #1965. Please edit if needed.

### CODE REVIEW

**Comment by endrift** [comment]
I don't get it. This removes functionality that was added intentionally, and is also not the correct fix.

**Comment by Testsr** [comment]
What functionality, exactly?

**Comment by endrift** [comment]
The one where you can actually set the default value? As opposed to just baking it into the UI, which is only there as a guideline, not as the ground truth.

Not to mention the fact that it has not one, but three style guide violations.

**Comment by Testsr** [comment]
The one where you can actually set the default value?
 Yes, you can still do that.   
`loadSetting("fastForwardMute", m_ui.muteFf, m_ui.mute->isChecked());`

**Comment by Testsr** [comment]
I should maybe name the commit better.

**Comment by Testsr** [comment]
Not to mention that the guidelines

**Comment by Testsr** [comment]
@endrift, What style guides violation, are there?

File: src/platform/qt/SettingsView.cpp [hunk 1]
```diff
@@ -563,8 +563,8 @@ void SettingsView::reloadConfig() {
 	loadSetting("gb.bios", m_ui.gbBios);
 	loadSetting("gbc.bios", m_ui.gbcBios);
 	loadSetting("sgb.bios", m_ui.sgbBios);
-	loadSetting("sgb.borders", m_ui.sgbBorders, true);
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:566]: This should not be removed; it's for setting the default when no option is present and entirely unrelated to the bug in question. Why did you remove it?

File: src/platform/qt/SettingsView.cpp [hunk 2]
```diff
@@ -563,8 +563,8 @@ void SettingsView::reloadConfig() {
...
-	loadSetting("useCgbColors", m_ui.useCgbColors, true);
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:567]: This should not be removed; it's for setting the default when no option is present and entirely unrelated to the bug in question. Why did you remove it?

File: src/platform/qt/SettingsView.cpp [hunk 3]
```diff
@@ -577,9 +577,9 @@ void SettingsView::reloadConfig() {
 	loadSetting("lockAspectRatio", m_ui.lockAspectRatio);
 	loadSetting("lockIntegerScaling", m_ui.lockIntegerScaling);
 	loadSetting("interframeBlending", m_ui.interframeBlending);
-	loadSetting("showOSD", m_ui.showOSD, true);
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:580]: This should not be removed; it's for setting the default when no option is present and entirely unrelated to the bug in question. Why did you remove it?

File: src/platform/qt/SettingsView.cpp [hunk 4]
```diff
@@ -577,9 +577,9 @@ void SettingsView::reloadConfig() {
...
+	loadSetting("showOSD", m_ui.showOSD);
 	loadSetting("volume", m_ui.volume, 0x100);
-	loadSetting("mute", m_ui.mute, false);
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:582]: This should not be removed; it's for setting the default when no option is present and entirely unrelated to the bug in question. Why did you remove it?

File: src/platform/qt/SettingsView.cpp [hunk 5]
```diff
@@ -596,12 +596,12 @@ void SettingsView::reloadConfig() {
 	loadSetting("cheatsPath", m_ui.cheatsPath);
 	loadSetting("showLibrary", m_ui.showLibrary);
 	loadSetting("preload", m_ui.preload);
-	loadSetting("showFps", m_ui.showFps, true);
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:599]: This should not be removed; it's for setting the default when no option is present and entirely unrelated to the bug in question. Why did you remove it?

File: src/platform/qt/SettingsView.cpp [hunk 6]
```diff
@@ -596,12 +596,12 @@ void SettingsView::reloadConfig() {
...
-	loadSetting("cheatAutoload", m_ui.cheatAutoload, true);
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:600]: This should not be removed; it's for setting the default when no option is present and entirely unrelated to the bug in question. Why did you remove it?

File: src/platform/qt/SettingsView.cpp [hunk 7]
```diff
@@ -596,12 +596,12 @@ void SettingsView::reloadConfig() {
...
-	loadSetting("cheatAutosave", m_ui.cheatAutosave, true);
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:601]: This should not be removed; it's for setting the default when no option is present and entirely unrelated to the bug in question. Why did you remove it?

File: src/platform/qt/SettingsView.cpp [hunk 8]
```diff
@@ -596,12 +596,12 @@ void SettingsView::reloadConfig() {
...
-	loadSetting("showFilename", m_ui.showFilename, false);
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:602]: This should not be removed; it's for setting the default when no option is present and entirely unrelated to the bug in question. Why did you remove it?

File: src/platform/qt/SettingsView.cpp [hunk 9]
```diff
@@ -596,12 +596,12 @@ void SettingsView::reloadConfig() {
...
-	loadSetting("autoload", m_ui.autoload, true);
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:603]: This should not be removed; it's for setting the default when no option is present and entirely unrelated to the bug in question. Why did you remove it?

File: src/platform/qt/SettingsView.cpp [hunk 10]
```diff
@@ -596,12 +596,12 @@ void SettingsView::reloadConfig() {
...
-	loadSetting("autosave", m_ui.autosave, false);
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:604]: This should not be removed; it's for setting the default when no option is present and entirely unrelated to the bug in question. Why did you remove it?

File: src/platform/qt/SettingsView.cpp [hunk 11]
```diff
@@ -735,7 +735,12 @@ void SettingsView::saveSetting(const char* key, const QVariant& field) {
 	m_controller->setOption(key, field);
 	m_controller->updateOption(key);
 }
-
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:738]: Line break needed between functions.

Why did you add a version of this function that doesn't have a default setting? The UI is just a guideline and does not hold what the ground truth for these default should be, but with this change that's where the defaults are coming from.

File: src/platform/qt/SettingsView.cpp [hunk 12]
```diff
@@ -735,7 +735,12 @@ void SettingsView::saveSetting(const char* key, const QVariant& field) {
...
+void SettingsView::loadSetting(const char* key, QAbstractButton* field) {
+	QString option = loadSetting(key);
+	if(option.isNull())
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:740]: Space between `if` and the `(`

File: src/platform/qt/SettingsView.cpp [hunk 13]
```diff
@@ -735,7 +735,12 @@ void SettingsView::saveSetting(const char* key, const QVariant& field) {
...
+		return;
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:741]: No bare statements. Needs braces.

File: src/platform/qt/SettingsView.cpp [hunk 14]
```diff
@@ -735,7 +735,12 @@ void SettingsView::saveSetting(const char* key, const QVariant& field) {
...
+	field->setChecked(option != "0");
+}
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:743]: Line break needed between functions.

**Review by endrift — CHANGES_REQUESTED** [review]
Ok, I pointed out the style nits, as well as every unrelated and seemingly pointless (as well as breaking) change you made in this PR.

It's every line.

File: src/platform/qt/SettingsView.cpp [hunk 15]
```diff
@@ -735,6 +735,13 @@ void SettingsView::saveSetting(const char* key, const QVariant& field) {
 	m_controller->setOption(key, field);
 	m_controller->updateOption(key);
 }
+void SettingsView::loadSetting(const char* key, QAbstractButton* field) {
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:738]: Still need a line break here.

File: src/platform/qt/SettingsView.cpp [hunk 16]
```diff
@@ -735,6 +735,13 @@ void SettingsView::saveSetting(const char* key, const QVariant& field) {
...
+	QString option = loadSetting(key);
+	if(option.isNull()){
```
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:740]: Still need that space between `if` and `(`, and now it also needs the space between `)` and `{`.

If you want to contribute to this project, or really any other project, you should take a look at surrounding code and match its style. There's a contributing.md that contains some very basic style rules that were not followed here.

**Review by endrift — CHANGES_REQUESTED** [review]
You didn't resolve all of the nits I pointed out last time and you added a new issue...

File: src/platform/qt/SettingsView.cpp (hunk 11 above)
Comment (Testsr) [reply src/platform/qt/SettingsView.cpp:738]: Why should the UI be just a guideline? The only example that would warrant overriding it, IMO would be fastFowardMute.

Comment (endrift) [reply src/platform/qt/SettingsView.cpp:738]: Why would the UI not be just a guideline? It's a terrible way to view defaults since you need a visual editor or skimming through lots and lots of XML, and is several levels detached from the actual names of the settings. The actual defaults are dictated by code, and this SettingsView class doesn't even dictate what the defaults are where they're actually used: it already needs to be kept in sync. Making the UI need to be kept in sync too is just a headache and hard to debug.

File: src/platform/qt/SettingsView.cpp (hunk 2 above)
Comment (endrift) [inline src/platform/qt/SettingsView.cpp:567]: You left out the comma when you "reverted" this.

**Review by endrift — CHANGES_REQUESTED** [review]
You still haven't told me what motivates this change; it still migrates defaults to the UI XML from this file on the offhand chance that they differ, which now means another place to look when something goes wrong, increasing maintenance burden. It doesn't actually fix any bugs in the process, either, despite your claim: it just, as I said, migrated the default, which coincidentally fixed it, instead of setting the default the way that's done within the framework you're changing.

**Comment by Testsr** [comment]
> You still haven't told me what motivates this change; it still migrates defaults to the UI XML from this file on the offhand chance that they differ, which now means another place to look when something goes wrong, increasing maintenance burden. It doesn't actually fix any bugs in the process, either, despite your claim: it just, as I said, migrated the default, which coincidentally fixed it, instead of setting the default the way that's done within the framework you're changing.

It does not "migrates defaults to the UI XML from this file on the offhand chance that they differ", instead it migrates defaults if this file does not provide a default.

**Comment by endrift** [comment]
That's not a meaningful distinction. You've yet to actually defend this design change, and at this point I don't think you have a reason for it other than you changed a thing to see if it worked and it did without realizing what the change did or why it worked.

Also I fixed the bug already without changing the design.

**Comment by Testsr** [comment]
The change are just for the cases where you forget to set a item to true.

**Comment by Testsr** [comment]
I can however leave this failed pull request behind.

**Comment by Testsr** [comment]
> That's not a meaningful distinction. You've yet to actually defend this design change, and at this point I don't think you have a reason for it other than you changed a thing to see if it worked and it did without realizing what the change did or why it worked.
> 
> Also I fixed the bug already without changing the design.
This change helps prevent the cases where you forget to set a new setting's default to true and therefore the function sets it to false by default. Having two places to look at can be a problem, however, so this is fine closed.
